### ISSUE TITLE
Unable to produce StochRSI with TEMA as a source.

### ISSUE THREAD

**Issue by cyberyaga** [DaveSkender/Stock.Indicators#849]
Hello... I've been working with this component a couple of months now. I've built my project around it. It's great.

I am having a bit of difficulty trying to replicate an indicator on my chart.

I need to create a StochRSI (1,1,5,2) with the source of it's data from TEMA(9). _I'm only interested in the K of the StochRSI, which produces 0 and 100 only._

Here is what it looks like in TradingView:
![image](https://user-images.githubusercontent.com/30553591/179905925-8cdd65fe-3013-4a7e-85e7-348218bde9ca.png)

My problem is that when I try to do this. I only get 0 and nothing else. I'm supposed to get 100 for certain candles.

Here is my code:
```csharp     
 //convert candles
            var quotes = s.CandleHistories.Select(c => new Quote()
                                            {
                                                Date = c.Time.DateTime,
                                                Open = c.Open,
                                                High = c.High,
                                                Low = c.Low,
                                                Close = c.Close,
                                                Volume = c.Volume
                                            });
            
            //Get tema values
            var tema9 = quotes.GetTema(9);

            //Get StochRSI with Tema9 as source
            var temaquotes = tema9
                                .Where(x => x.Tema.HasValue)
                                .Select(x => new Quote() { 
                                    Close = Convert.ToDecimal(x.Tema.Value),
                                    Date = x.Date
                                });

            //This output ONLY produces 0... no 100s
            var stochrsiTema = temaquotes.GetStochRsi(1,1,5,2);
```

**Comment by cyberyaga** [comment]
Thanks Dave for your response. 

And Yeah... I came to the sad realization that TradingView is not reliable as a comparison source, because it's data it's garbage and they sometimes do not so evident things with their indicators behind the scene, that are not obvious. -_-

The `tema9` object has valid TEMA data in it. And That's why I'm doing a `Where` on the data when converting it to `IEnumerable<Quote>`.

BTW... This: `quotes.GetTema(9).GetStochRsi(1,1,5,2);` gives an error:
`The type 'Skender.Stock.Indicators.TemaResult' cannot be used as type parameter 'TQuote' in the generic type or method 'Indicator.GetStochRsi<TQuote>(IEnumerable<TQuote>, int, int, int, int)'. There is no implicit reference conversion from 'Skender.Stock.Indicators.TemaResult' to 'Skender.Stock.Indicators.IQuote'.`
That's why I'm doing the multi-step process.

I guess I can try to figure out what's going on by downloading the source and trying to run this in debug and see why it always goes to `0`.

**Comment by DaveSkender** [comment]
I might be able to look at it in a couple weeks ☹️, to better explain what's happening, if you send me your base quotes, but can't promise anything soon.  I suspect you'll have moved on by then.

**Comment by cyberyaga** [comment]
Hello... I've been on and off trying to figure this out. I've been focusing on other things in this particular project.

Here is what the candle data I'm using looks like:
![image](https://user-images.githubusercontent.com/30553591/180923657-754a7331-d184-4437-a535-5dba1eb84b6f.png)
I've also attached the candle data I'm using. This is ETCBTC in case you were curious.
[candledata.csv](https://github.com/DaveSkender/Stock.Indicators/files/9186530/candledata.csv)

I was wondering if maybe the fact that the numbers are so small (ex. 0.000715) was getting zeroed out in a conversion inside the indicator's code. So, I tried moving the decimal point so that there were some whole numbers. But that didn't work. All of the StochRSI values were zero.

I tried walking through your code, but the project is now in .NET 7. I installed .NET 7 on my machine and it messed up my current .NET setup, so I had to undo that, in order to be able to develop.

Again... Thanks Dave for your help and for making this project. I will try to install .NET 7 again and see if I'm able to walk through the code.

And sure, if you are only able to look at this in a couple of weeks, it's fine... I take what I can get :)

Thanks

**Comment by DaveSkender** [comment]
I’ll try to take a look this weekend.  Thanks for the extra info.

> the project is now in .NET 7 …

I usually wait for GA before taking on new .NET versions, but got ahead of myself on this one.  I sorta regret it as it requires me to keep two versions of Visual Studio on my computer and it breaks some VS Code IDE compatibility.  Probably won’t do it again.

**Comment by DaveSkender** [comment]
I think the error in your use case here is a simple "oopsie".  All you need to do is correct the order of your parameter arguments.

```csharp
var results = quotes
    .GetTema(9)
    .GetStochRsi(5,2,1,1);
```
Chaining like this is now available in v2.1.1 for Stochastic RSI

**Comment by cyberyaga** [comment]
Oh Wow... What a stupid mistake... 

Thanks for the help again. At least we got the GetStochRsi method to chain with GetTema :)
